api/v1/routes/user.py

```diff
@@ -0,0 +1,27 @@
+from fastapi import Depends, HTTPException, APIRouter, Request
+from sqlalchemy.orm import Session
+
+from ..models.user import User
+from api.v1.schemas.user import DeactivateUserSchema
+from api.db.database import get_db
+from api.utils.dependencies import get_current_user
+
+
+user = APIRouter(prefix='/api/v1/users', tags=['Users'])
+
+@user.patch('/accounts/deactivate', status_code=200)
+async def deactivate_account(request: Request, schema: DeactivateUserSchema, db: Session = Depends(get_db), user: User = Depends(get_current_user)):
+    '''Endpoint to deactivate a user account'''
+
+    if not user.is_active:
+        raise HTTPException(400, 'User is inactive')
+    
+    if schema.confirmation == False:
+        raise HTTPException(400, 'Confirmation required to deactivate account')
+
```

Comment:
A proper confirmation should be done by sending the user an email with a deactivation link

Boss @zxenonx what do you think?

Reply:
I had that before but the person working on adding the emailing template to the database doesn't have it ready so I was told to implement what I had, that a central service would be used for emailing
The endpoint to even snd the email was available but the person too could not proceed without the one adding the templates to the database

Reply:
I was the one who said you should implement what you have. Anyway, I hope the confirmation would allow the user revert during a set time, since you have decided to deactivate successfully before sending a confirmation

Reply:
Okay
Let me see what I can do about that as quickly as possible

Reply:
@Dev-wonderful  I have implemented the changes